app/src/main/java/com/example/weather/ChooseCity.java

```diff
@@ -0,0 +1,131 @@
+package com.example.weather;
+
+import android.content.Intent;
+import android.os.PersistableBundle;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.CheckBox;
+import android.widget.EditText;
+import android.widget.Spinner;
+import android.widget.Toast;
+
+public class ChooseCity extends AppCompatActivity {
+
+    private Singleton single = Singleton.getInstance();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_choose_city);
+
+        final Spinner spinner = findViewById(R.id.choose_popular_city);
+        final EditText editCity = findViewById(R.id.edit_city);
+        final CheckBox checkBox = findViewById(R.id.checkBox);
+        final Intent getWeatherIntent = new Intent(this, MainActivity.class);
+        Button showWeather = findViewById(R.id.show_weather);
+
+        showWeather.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+
+                if (editCity.getText().toString().matches("")) {
```

Comment:
тут надо разделить заполнение интента и обработку спиннера

```suggestion
               String paramCity;
                if (editCity.getText().toString().matches("")) {
                       paramCity = String.valueOf(spinner.getSelectedItem());
                       editCity.setText(paramCity );
                  } else {
                       paramCity = editCity.getText().toString();
                  }
                  
                  Intent intent = new Intent(getContext(), MainActivity.class);
                  intent.putExtra(MainActivity.NAME_CITY_SPINNER, paramCity);
                  intent.putExtra(MainActivity.NAME_CITY_SPINNER, checkBox.isChecked());
                  startActivity(Intent);
```

Примерно так

Reply:
И еще лучше, вынести эту обработку  в отдельный метод, слишком длинный метод

Reply:
Ага, понял. Переработал и вынес в отдельный метод